changelog/3290.bugfix

```diff
@@ -0,0 +1,2 @@
+Improved `monkeypatch` to support some form of with statement. Now you can use `with monkeypatch.context() as m:`
```

Comment:
This is a new feature so it should be named `3290.feature` instead. Also I suggest using this text:

```rst
``monkeypatch`` now supports a ``context()`` function which acts as a context manager which undoes all patching done within the ``with`` block.
```

The fact that it can be used to avoid blowing up pytest itself is not the highlight of the new feature, just a recommended workaround.